sdk/python/foundation-models/system/inference/text-to-image/text-to-image-batch-endpoint.ipynb

```diff
@@ -0,0 +1,360 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Set the deployment as default"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "endpoint = workspace_ml_client.batch_endpoints.get(endpoint_name)\n",
+    "endpoint.defaults.deployment_name = deployment_name\n",
+    "workspace_ml_client.begin_create_or_update(endpoint).result()\n",
+    "\n",
+    "endpoint = workspace_ml_client.batch_endpoints.get(endpoint_name)\n",
+    "print(f\"The default deployment is {endpoint.defaults.deployment_name}\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 4 Test the endpoint - Using a csv file having a list of text prompts\n",
+    "\n",
+    "Invoke the batch endpoint with the input parameter pointing to the csv file containing the batch inference input. This creates a pipeline job using the default deployment in the endpoint. Wait for the job to complete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "dataset_parent_dir = \"batch_data\"\n",
+    "csv_file_path = os.path.join(\"batch_data\", \"text_prompt_list.csv\")\n",
+    "\n",
+    "job = None\n",
+    "input = Input(path=csv_file_path, type=AssetTypes.URI_FILE)\n",
+    "num_retries = 3\n",
+    "for i in range(num_retries):\n",
+    "    try:\n",
+    "        job = workspace_ml_client.batch_endpoints.invoke(\n",
+    "            endpoint_name=endpoint.name, input=input\n",
+    "        )\n",
+    "        break\n",
+    "    except Exception as e:\n",
+    "        if i == num_retries - 1:\n",
+    "            raise e\n",
+    "        else:\n",
+    "            print(\"Endpoint invocation failed. Retrying after 5 seconds...\")\n",
+    "            time.sleep(5)\n",
+    "if job is not None:\n",
+    "    workspace_ml_client.jobs.stream(job.name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "scoring_job = list(workspace_ml_client.jobs.list(parent_job_name=job.name))[0]\n",
+    "\n",
+    "workspace_ml_client.jobs.download(\n",
+    "    name=scoring_job.name,\n",
+    "    download_path=os.path.join(dataset_parent_dir, \"generated_images\"),\n",
+    "    output_name=\"score\",\n",
+    ")\n",
+    "\n",
+    "predictions_file = os.path.join(\n",
+    "    dataset_parent_dir, \"generated_images\", \"named-outputs\", \"score\", \"predictions.csv\"\n",
+    ")\n",
+    "\n",
+    "# Load the batch predictions file with no headers into a dataframe and set your column names\n",
+    "score_df = pd.read_csv(\n",
+    "    predictions_file,\n",
+    "    header=None,\n",
+    "    names=[\"prompt\", \"image_file_name\", \"nsfw_content_detected\", \"input_csv_name\"],\n",
```

Comment:
first column generally consist of index. Please recheck this.

Reply:
I have verified this. Index comes separately

Reply:
Can you please confirm this with final models.